### PR TITLE
FIX UTF8 handling in `kassenbuch client create`

### PR DESCRIPTION
**Pull request by mgmax**

### CODE REVIEW

File: FabLabKasse/kassenbuch.py [hunk 1]
```diff
@@ -1277,19 +1277,19 @@ def fetch_input(explanation, default_input=None,
                 :type allowed_regexp: basestr | None
                 :type extra_checks: function | None
                 """
-                default_str = " [{0}]".format(unicode(default_input)) if default_input is not None else ""
+                default_str = u" [{0}]".format(unicode(default_input)) if default_input is not None else ""
```
Comment (patkan) [inline FabLabKasse/kassenbuch.py:1280]: Shouldn't the second `""` be also `u""`?

File: FabLabKasse/kassenbuch.py [hunk 2]
```diff
@@ -1277,19 +1277,19 @@ def fetch_input(explanation, default_input=None,
...
                 allowed_regexp = allowed_regexp if allowed_regexp is not None else ur".*"
                 extra_checks = extra_checks if extra_checks else lambda x: True
 
                 while True:
                     input_str = raw_input(u"{e}{d}: ".format(
-                        e=explanation, d=default_str))
+                        e=explanation, d=default_str)).decode(sys.stdin.encoding)
 
                     if input_str == '' and default_input is not None:
```
Comment (patkan) [inline FabLabKasse/kassenbuch.py:1288]: I am not sure whether comparison with an empty string is the same with and without unicode.  (is `u""` == `""`)

OK, i just checked it: both in Python 2 and 3 `'' == u''` evaluates to `True`

**Comment by patkan** [comment]
I just tested your branch with the conflicting data from the mail thread (it contained umlauts in the comment).
`development` crashes with the data and this branch creates the entry. I have tested `client list` and `client show`, both work still after the fix :-)
The entry in the database looks good and both umlauts in address and comment do work now (and failed in `development`)
Booking onto the customer in the GUI works too.

Concerning my first comment about the empty string that should be unicode:
I had a look at the code. Since the string gets inserted into a unicode-string with format it is irrelevant whether the empty string is unicode or not.
For completeness I would be happy, if you could change the second string to unicode.

```
[pseudo-patch]
-                default_str = u" [{0}]".format(unicode(default_input)) if default_input is not None else ""
+                default_str = u" [{0}]".format(unicode(default_input)) if default_input is not None else u""
```

Apart from that I would say this fix is ready to merge.

**Comment by mgmax** [comment]
Feel free to edit and merge :-)

**Comment by patkan** [comment]
merged ;-)

**Comment by patkan** [comment]
(and deployed)

**Pull request closed without merging**